### ISSUE TITLE
Variables is not resolved when name non-case-sensetively matches Type 

### ISSUE THREAD

**Issue by eliasku** [TiVo/intellij-haxe#405]
If name of variable match Type-name is **non-case-sensetive** mode - resolving is broken.
If name is different - all works fine.
To reproduce - just declare variables `var string:String;`, `var xml:Xml`, `var array:Array<Int>` - all broken. Also broken for method's arguments.

Issue has a very negative impact on the intellij-haxe user experience.

Version: next 0.9.9
OSX, IntelliJ IDEA 14.1, 15

![image](https://cloud.githubusercontent.com/assets/3038174/12866336/b2c7d782-ccdb-11e5-869c-ce77f9fa5b8b.png)

Repro:
```
	public function foo(arr:Array<Int>, array:Array<Int>) {
		//array.<caret> ---> doesn't work

		// BUT:
		//arr.<caret>  ---> works
		// rename `array` to `arr` as workaround
	}

	public static function test(string:String, str:String) {
		//string.<caret>
		//str.<caret> -- works

		var xml:Xml = Xml.parse(content); // valid
		var root:Xml = xml.firstElement(); // `firstElement` not resolved

		// no completions and symbol not resolved
		var date = Date.now();
		var year = toExpr(date.getFullYear());
		var month = toExpr(date.getMonth());
		var day = toExpr(date.getDate());
		var hours = toExpr(date.getHours());
		var mins = toExpr(date.getMinutes());
		var secs = toExpr(date.getSeconds());
	}
```

**Comment by as3boyan** [comment]
I know a limited solution for this, like as you said check if item text matches filename in a case sensetive way. It should help in most cases.

**Comment by as3boyan** [comment]
Also this is duplicate to #234

**Comment by eliasku** [comment]
Oh, sorry for the duplicate. I'm going to make PR with fix (using your solution), and cover issue with additional unit-tests

**Comment by eliasku** [comment]
here we go - https://github.com/TiVo/intellij-haxe/pull/406

**Comment by as3boyan** [comment]
This issue might mean that we have wrong type resolution order, but if it is fixes most cases - that is good

**Comment by EBatTiVo** [comment]
Fixed PR #406: d44634b
